tests/network/services.go

```diff
@@ -247,51 +237,35 @@ var _ = SIGDescribe("Services", func() {
 		})
 
 		Context("with a service matching the vmi exposed", func() {
-			var jobsToCleanup []func() error
-			var serviceManager *vmiServiceManager
-			var serviceNamePrefix string
-
-			BeforeEach(func() {
-				serviceNamePrefix = "myservice"
-				serviceManager = newVMIServiceManager(inboundVMI, servicePort, serviceNamePrefix, selectorLabelKey, selectorLabelValue)
-
-				services := serviceManager.buildK8sServicesSpec()
-				Expect(services).NotTo(BeEmpty(), "a service should be exposed per each iface. At least one *must* be present.")
-
-				for _, exposedService := range services {
-					_, err := virtClient.CoreV1().Services(inboundVMI.Namespace).Create(&exposedService)
-					Expect(err).NotTo(HaveOccurred())
-				}
-
-				// reset the job cleanup functions list
-				jobsToCleanup = nil
-			})
+			var jobToCleanup func() error
 
 			AfterEach(func() {
-				var errors []error
-				for _, exposedService := range serviceManager.services {
-					err := virtClient.CoreV1().Services(inboundVMI.Namespace).Delete(exposedService.Name, &k8smetav1.DeleteOptions{})
-					errors = append(errors, err)
-				}
-				for _, err := range errors {
-					Expect(err).NotTo(HaveOccurred())
-				}
+				Expect(jobToCleanup()).To(Succeed(), "cleaning up the k8sv1.Service entity should have succeeded.")
```

Comment:
You should also delete the service.

Reply:
Right! 

Done.

I'm not sure I'm doing the service cleanup in a correct way (I've just added it to the `describeTable` body ... That seemed less wrong that cleaning it up in the after each **if** the service exists (since on the next test, we do not create the service). Your thoughts @AlonaKaplan / @EdDev ?